### ISSUE TITLE
calculate actual altitude from gpsalt and pressure

### ISSUE THREAD

**Issue by hgloeckner** [atmdrops/pydropsonde#69]
with aspen somehow moving altitude, it would be good to calculate altitude over MSL ourselves.

**Comment by hgloeckner** [comment]
this won't be done for now since we switched to gpsalt as our height dimension

**Comment by tmieslinger** [comment]
An update and further discussion points on the altitude problem:
* we learnt so far that the GPS altitude given by the dropsondes is indeed altitude above MSL.
* in general, the GPS module has more missing values especially in the lower atmosphere compared to the PTU module
* the "pressure altitude" derived from the PTU module assumes that the sonde hit the surface. This is in a handful of cases from Orcestra not true and leads to a shifted and wrong pressure profile.

To make most out of the altitude information (save as many sondes as possible) we could take advantage of the two independent measurements with the following priority:
1. we use the pressure `alt` per default together with an additional test that uses the difference in `gpsalt`and `alt` as an indicator for a shifted profile and thus, use `alt`for sondes that measure until the surface.
2.  we might call the above mentioned test `sonde_hit_surface` (following the Aspen-style ) and if it fails, we can try to derive the difference in `gpsalt`and `alt`and shift the pressure altitude profile up by adding the offset.
3. we use `gpsalt`whenever the PTU module failed completely
4. we save the corrected/combined altitude product to a new variable, maybe `altitude`, and we might want to add an "auxiliary" variable `altitude_source` that gives the source of the altitude values for a given sonde. **I would not mix altitude sources within a single sonde**.

The `sonde_hit_surface` (boolean) test could use the infromation from a function called e.g. `offset_gpsalt_alt` that returns a numeric value representing the difference or shift in the profiles. For that shift we basically need one or more points at same altitude to compare. Due to the different sampling frequencies and due to GPS signal failing more often in the lowest part of the atmosphere, I would suggest to use a relatively large range of e.g. 2-4km and calculate the mean difference for all points that are not NaN in both variables. @hgloeckner that should be very close to the current `gpsalt_near_alt` test.

**Comment by hgloeckner** [comment]
I agree, that's something to work on - I think the alt-near-gpsalt test in the current form does not help here, because it is prone to errors if gpsalt is missing for a 100m at either and of whatever interval is chosen. Then the whole test would be negative although it does not necessarily mean that the two measure different things. 

My current idea would be to 
 1. add a `p_low_physics` test that checks if the last measured pressure is within a given range (for ORCESTRA something like 1006 and 1014 hPa). (PR #154 )
2. change the `near_surface` test to use `gpsalt` for `u` and `v` and `alt` for `rh`, `ta` and `p`. 
3. use `gpsalt` as the default height coordinate, because there are no more assumptions going into it apart from the actual measurement
4. if `u_near_surface` is `False` (i.e. not enough surface gps-measurements), AND `p_low_physics` is `True`, then the altitude is changed to `alt`, because probably gps was lost somewhere.
5. if there is no `gpsalt` at all and the `p_low_physics` is `False` a sonde would be dropped.

As you suggested, I think that a new altitude dimension should be put in place and the old ones should be dropped as to avoid confusion and usage of wrong altitudes. 

I tried to implement something like this, and I think I can make a PR later today, so that we could move the discussion? 
Also, `alt` could be the default altitude dimension, however, I think @lkluft pointed out, that there are a bit more assumptions going in there than in `gpsalt`. For the latter, we know that the error is 10-20m, but apart from that it's measurement, while the geopotential altitude calculation needs hydrostaticity (I think).
